### ISSUE TITLE
`Invalid owner ID` when updating to Craft 3.5

### ISSUE THREAD

**Issue by jerome2710** [spicywebau/craft-neo#396]
#### Description
We currently have > 20 Craft sites running on 3.4 with the Neo plugin installed. Now 3.5 is officially released and a few bug fixes have been made, we are looking to upgrade all our sites to 3.5. Unfortunately, we are experiencing this issue on each and every site we are trying to migrate.

```
2020-08-21 10:37:50 [-][-][-][error][yii\base\InvalidConfigException] yii\base\InvalidConfigException: Invalid owner ID: 9185 in /.../vendor/spicyweb/craft-neo/src/elements/Block.php:282
Stack trace:
#0 /.../vendor/craftcms/cms/src/helpers/ElementHelper.php(385): benf\neo\elements\Block->getOwner()
#1 /.../vendor/craftcms/cms/src/helpers/ElementHelper.php(385): craft\helpers\ElementHelper::rootElement(Object(benf\neo\elements\Block))
#2 /.../vendor/craftcms/cms/src/helpers/ElementHelper.php(399): craft\helpers\ElementHelper::rootElement(Object(craft\elements\MatrixBlock))
#3 /.../vendor/craftcms/cms/src/services/Elements.php(1411): craft\helpers\ElementHelper::isDraftOrRevision(Object(craft\elements\MatrixBlock))
#4 /.../vendor/craftcms/cms/src/fields/Matrix.php(1024): craft\services\Elements->deleteElement(Object(craft\elements\MatrixBlock), false)
#5 /.../vendor/craftcms/cms/src/base/Element.php(3200): craft\fields\Matrix->beforeElementDelete(Object(benf\neo\elements\Block))
#6 /.../vendor/spicyweb/craft-neo/src/elements/Block.php(478): craft\base\Element->beforeDelete()
#7 /.../vendor/craftcms/cms/src/services/Elements.php(1392): benf\neo\elements\Block->beforeDelete()
#8 /.../vendor/spicyweb/craft-neo/src/services/BlockTypes.php(453): craft\services\Elements->deleteElement(Object(benf\neo\elements\Block))
#9 /.../vendor/craftcms/cms/src/services/ProjectConfig.php(1089): benf\neo\services\BlockTypes->handleDeletedBlockType(Object(craft\events\ConfigEvent))
#10 [internal function]: craft\services\ProjectConfig->handleChangeEvent(Object(craft\events\ConfigEvent))
#11 /.../vendor/yiisoft/yii2/base/Component.php(627): call_user_func(Array, Object(craft\events\ConfigEvent))
#12 /.../vendor/craftcms/cms/src/services/ProjectConfig.php(661): yii\base\Component->trigger('removeItem', Object(craft\events\ConfigEvent))
#13 /.../vendor/craftcms/cms/src/services/ProjectConfig.php(1079): craft\services\ProjectConfig->processConfigChanges('neoBlockTypes.9...')
#14 [internal function]: craft\services\ProjectConfig->handleChangeEvent(Object(craft\events\ConfigEvent))
#15 /.../vendor/yiisoft/yii2/base/Component.php(627): call_user_func(Array, Object(craft\events\ConfigEvent))
#16 /.../vendor/craftcms/cms/src/services/ProjectConfig.php(661): yii\base\Component->trigger('removeItem', Object(craft\events\ConfigEvent))
#17 /.../vendor/craftcms/cms/src/services/ProjectConfig.php(1179): craft\services\ProjectConfig->processConfigChanges('neoBlockTypes.9...', false, NULL, true)
#18 /.../vendor/craftcms/cms/src/services/ProjectConfig.php(527): craft\services\ProjectConfig->_applyChanges(Array)
#19 /.../vendor/craftcms/cms/src/console/controllers/ProjectConfigController.php(97): craft\services\ProjectConfig->applyYamlChanges()
#20 [internal function]: craft\console\controllers\ProjectConfigController->actionApply()
#21 /.../vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#22 /.../vendor/yiisoft/yii2/base/Controller.php(180): yii\base\InlineAction->runWithParams(Array)
#23 /.../vendor/yiisoft/yii2/console/Controller.php(181): yii\base\Controller->runAction('apply', Array)
#24 /.../vendor/craftcms/cms/src/console/Controller.php(188): yii\console\Controller->runAction('apply', Array)
#25 /.../vendor/craftcms/cms/src/console/controllers/ProjectConfigController.php(119): craft\console\Controller->runAction('apply')
#26 [internal function]: craft\console\controllers\ProjectConfigController->actionSync()
#27 /.../vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#28 /.../vendor/yiisoft/yii2/base/Controller.php(180): yii\base\InlineAction->runWithParams(Array)
#29 /.../vendor/yiisoft/yii2/console/Controller.php(181): yii\base\Controller->runAction('sync', Array)
#30 /.../vendor/craftcms/cms/src/console/Controller.php(188): yii\console\Controller->runAction('sync', Array)
#31 /.../vendor/yiisoft/yii2/base/Module.php(528): craft\console\Controller->runAction('sync', Array)
#32 /.../vendor/yiisoft/yii2/console/Application.php(180): yii\base\Module->runAction('project-config/...', Array)
#33 /.../vendor/craftcms/cms/src/console/Application.php(87): yii\console\Application->runAction('project-config/...', Array)
#34 /.../vendor/yiisoft/yii2/console/Application.php(147): craft\console\Application->runAction('project-config/...', Array)
#35 /.../vendor/yiisoft/yii2/base/Application.php(386): yii\console\Application->handleRequest(Object(craft\console\Request))
#36 /.../craft(23): yii\base\Application->run()
#37 {main}
```

This seem to be caused by https://github.com/spicywebau/craft-neo/blob/master/src/elements/Block.php#L282, when the owning element of the block cannot be found. However, while running a query across all blocks, there are no blocks where the element seems to be missing:

```
SELECT craft_neoblocks.ownerId, craft_elements.id FROM craft_neoblocks LEFT JOIN craft_elements ON craft_neoblocks.ownerId = craft_elements.id
```

Could this be fixed? We would like to update all our sites and this is now blocking the process.

#### Other information
- Neo version: 2.8.6
- Craft version: 3.5.5
- (if multi-site) What is the affected Neo field's propagation method: None / Only save blocks to the site they were created in
- Is eager-loading used: unsure

**Comment by genetschneider** [comment]
Ran into the same issue when updated craft-neo from `2.8.1` to `2.8.4`. Craft version we're using is `3.5.0`.
Our reproduction steps:
- Have multiple sites
- Update craft-neo version to `2.8.4`
- Go to create a new entry
- Once we add a child neo block, we get a `Invalid owner ID` error

**Comment by ttempleton** [comment]
@jerome2710, thanks for sending the files; unfortunately I could not reproduce the error with that setup, but after some more testing with my usual testing environment, I did find a situation where I could cause this error.  It appears that Neo blocks are not being deleted if the *site* they belong to is deleted or disabled for their entry's section.  A later attempt by the system to delete those blocks will then cause this error.  I'll keep working on it and hope to have a fix released soon.

**Comment by ttempleton** [comment]
I've made progress on this in the [`bugfix/delete-blocks-with-site`](https://github.com/spicywebau/craft-neo/tree/bugfix/delete-blocks-with-site) branch.  It seems to be working well so far for Neo fields that only save blocks to the site they were created in, but I'll still need to do more testing especially with the site group / language propagation methods before release.

**Comment by jordikroon** [comment]
@ttempleton do you have an update regarding the fix? We want to get our sites to Craft 3.5, therefore we need this fix.

**Comment by bryantAXS** [comment]
I just ran into this issue as well today. Would love to get an update!

**Comment by jerome2710** [comment]
@ttempleton could you provide us with an update on this matter? You mentioned the progress in the branch, but some difficulties with the propagations. When could this be fixed and released as an update?

We are paying $49 and $19/year for updates almost 20 times for our websites using your plugin. It would be nice if such bugs would be fixed within a month after reporting them.

**Comment by ttempleton** [comment]
This was a Craft bug caused by craftcms/cms#6780 and was fixed in Craft 3.5.9.
